src/transformers/modeling_utils.py

```diff
@@ -4095,6 +4111,26 @@ def from_pretrained(
                     f"Using `low_cpu_mem_usage=True` or a `device_map` requires Accelerate: `pip install 'accelerate>={ACCELERATE_MIN_VERSION}'`"
                 )
             init_contexts.append(init_empty_weights())
+        elif tp_plan is not None:
+            if not torch.distributed.is_initialized():
+                raise ValueError("Tensor Parallel requires torch.distributed to be initialized first.")
+
+            # Get device type (e.g. "cuda")
+            try:
+                # torch 2.6 API
+                device_type = torch.distributed.distributed_c10d._device_capability()[0]
+            except AttributeError:
+                if torch.cuda.is_available():
+                    device_type = "cuda"
+                else:
+                    raise RuntimeError(
+                        "Device type unknown. Please run model.tensor_parallel with an explicit DeviceMesh."
```

Comment:
I am not sure I see how this is actionnable? If you pass `tp_plan="auto"` but have a lower version of torch and no cuda how do you bypass this?

Reply:
let's make it more actionnable please!